keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,416 @@
+
+To work, this plugin requires topology information of the available resource on the worker nodes.
+
+Kubelet will be responsible for collecting all necessary resource information of the pods,
+based on allocatable resources on the node and allocated resources to pods. The NUMA nodes
+would be represented as Zones in Kubelet and the NodeResourceTopology would capture the 
+resource information at a zone level granularity.
+
+Once the information is captured in the NodeResourceTopology API, the scheduler can refer to
+it like it refers to Node Capacity and Allocatable while making a Topology-aware Scheduling decision.
+
+
+## Changes to the API
+
+Code responsible for working with NodeResourceTopology API will be placed in the stagingit g directory
+at path staging/src/k8s.io/api/node/v1/types.go.
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopology is a specification for a hardware resources
+type NodeResourceTopology struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TopologyPolicy []string `json:"topologyPolicies"`
+	Zones          ZoneMap  `json:"zones"`
+}
+
+// Zone is the spec for a NodeResourceTopology resource
+type Zone struct {
+    Name       string           `json:"name"`
+    Type       string           `json:"type"`
+    Parent     string           `json:"parent,omitempty"`
+    Costs      CostList         `json:"costs,omitempty"`
+    Attributes AttributeList    `json:"attributes,omitempty"`
+    Resources  ResourceInfoList `json:"resources,omitempty"`
+}
+
+type ResourceInfo struct {
+    Name        string `json:"name"`
+    Allocatable string `json:"allocatable"`
+    Capacity    string `json:"capacity"`
+}
+
+type ZoneList []Zone
+type ResourceInfoList []ResourceInfo
+
+type CostInfo struct {
+    Name  string `json:"name"`
+    Value int	  `json:"value"`
+}
+
+type AttributeInfo struct {
+    Name  string `json:"name"`
+    Value string `json:"value"`
+}
+
+type CostList []CostInfo
+type AttributeList []AttributeInfo
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopologyList is a list of NodeResourceTopology resources
+type NodeResourceTopologyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []NodeResourceTopology `json:"items"`
+}
+
+```
+
+
+Where TopologyPolicy may have following values: none, best-effort, restricted, single-numa-node.
+The current policies of TopologyManager can't coexist together at the same time, but in future such kind of policies could appear.
+For example we can have policy for HyperThreading and it can live with NUMA policies.
```

Comment:
I think this calls for a clear definition of what "policy" refers to, my understanding was that it is only about NUMA in this context.